ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestHelper.java

```diff
@@ -0,0 +1,137 @@
+import org.hibernate.ogm.persister.impl.OgmEntityPersister;
+import org.hibernate.ogm.utils.TestableGridDialect;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+
+/**
+ * @author Dmitriy Kozlov
+ */
+public class IgniteTestHelper implements TestableGridDialect {
+
+	@Override
+	public long getNumberOfEntities(SessionFactory sessionFactory) {
+		int entityCount = 0;
+		Set<IgniteCache<?, ?>> processedCaches = Collections.newSetFromMap( new IdentityHashMap<IgniteCache<?, ?>, Boolean>() );
+		for ( EntityPersister entityPersister : ( (SessionFactoryImplementor) sessionFactory ).getEntityPersisters().values() ) {
+			IgniteCache<?, ?> entityCache = getEntityCache( sessionFactory, ( (OgmEntityPersister) entityPersister ).getEntityKeyMetadata() );
+			if ( !processedCaches.contains( entityCache ) ) {
+				entityCount += entityCache.size( CachePeekMode.ALL );
+				processedCaches.add( entityCache );
+			}
+		}
+		return entityCount;
+	}
+
+	@Override
+	public long getNumberOfAssociations(SessionFactory sessionFactory) {
+		int associationCount = 0;
+		Set<IgniteCache<?, ?>> processedCaches = Collections.newSetFromMap( new IdentityHashMap<IgniteCache<?, ?>, Boolean>() );
+		for ( CollectionPersister colleactionPersister : ( (SessionFactoryImplementor) sessionFactory ).getCollectionPersisters().values() ) {
+			IgniteCache<?, ?> associationCache = getAssociationCache( sessionFactory, ( (OgmCollectionPersister) colleactionPersister ).getAssociationKeyMetadata() );
+			if ( !processedCaches.contains( associationCache ) ) {
+				associationCount += associationCache.size( CachePeekMode.ALL );
+				processedCaches.add( associationCache );
+			}
+		}
+		return associationCount;
+	}
+
+	@Override
+	public long getNumberOfAssociations(SessionFactory sessionFactory, AssociationStorageType type) {
+		return 0;
+	}
+
+	@Override
+	public Map<String, Object> extractEntityTuple(SessionFactory sessionFactory, EntityKey key) {
+		IgniteCache<String, BinaryObject> cache = getEntityCache( sessionFactory, key.getMetadata() );
+		String cacheKey = getProvider( sessionFactory ).getKeyProvider().getEntityKeyString( key );
+
+		Map<String, Object> result = new HashMap<>();
+		Object po = cache.get( cacheKey );
+
+		IgniteDialect igniteDialect = (IgniteDialect) ((SessionFactoryImplementor) sessionFactory).getServiceRegistry().getService( GridDialect.class );
+		TupleSnapshot snapshot = new IgniteDialect.IgnitePortableTupleSnapshot( po );
+		for (String fieldName : snapshot.getColumnNames()) {
+			result.put( fieldName, snapshot.get( fieldName ) );
+		}
+
+		return result;
+	}
+
+	@Override
+	public boolean backendSupportsTransactions() {
+		return true;
+	}
+
+	@Override
+	public void dropSchemaAndDatabase(SessionFactory sessionFactory) {
+		// TODO what to do here???
+	}
+
+	@Override
+	public Map<String, String> getEnvironmentProperties() {
+		// TODO and here?
```

Comment:
Unless you need to set some properties before running the tests, you can leave this method empty.
Maybe return an empty map.